src/rp2_common/pico_standard_link/new_delete.cpp

```diff
@@ -23,4 +23,6 @@ void operator delete(void *p) { std::free(p); }
 
 void operator delete[](void *p) noexcept { std::free(p); }
 
+void operator delete[](void *p, __unused std::size_t n) noexcept { std::free(p); }
```

Comment:
The C++ new header only adds the declarations for the sized deallocation functions if `__cpp_sized_deallocation` is true.

So, should the sized deallocation function that is being added with this PR (and the already existing size deallocation function on [line 20](https://github.com/raspberrypi/pico-sdk/blob/5446fa7ae922dd1aced35028799c4f615fd0dad9/src/rp2_common/pico_standard_link/new_delete.cpp#L20) be conditionally compiled using `__cpp_sized_deallocation` (see [here](https://en.cppreference.com/w/cpp/feature_test))?

Here is what the new header has:

```c++
#if __cpp_sized_deallocation
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
  __attribute__((__externally_visible__));
void operator delete[](void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
  __attribute__((__externally_visible__));
#endif
``´

Reply:
I think it will work either way. I looked up some other implementations of this fix and found one case where the conditional is included ([here](https://hg.mozilla.org/mozilla-central/rev/d346efc8449b)), and two where it was not ([here](https://hg.openjdk.java.net/jdk/jdk/rev/4ec9fc2b2f0d) and [here](https://github.com/cpputest/cpputest/issues/957)).

Reply:
I'm sure it will work too, however, I'm not sure if it's a good idea to make functions intended for C++14 and higher available in C++11.

Reply:
Updated